core/src/main/java/greencity/controller/HabitController.java

```diff
@@ -87,125 +88,171 @@
      *
-     * @param addHabitStatisticDto - dto for {@link HabitStatistic} entity.
-     * @return dto {@link AddHabitStatisticDto} instance.
-     * @author Yuriy Olkhovskyi.
+     * @param userVO             {@link UserVO} instance.
+     * @param id                 {@link Habit} id.
+     * @param habitAssignStatDto {@link HabitAssignStatDto} instance.
+     * @return {@link HabitAssignDto}.
      */
-    @ApiOperation(value = "Add habit statistic for assigned habit.")
+    @ApiOperation(value = "Update active user habit assign acquired or suspended status.")
     @ApiResponses(value = {
-        @ApiResponse(code = 201, message = HttpStatuses.CREATED, response = AddHabitStatisticDto.class),
-        @ApiResponse(code = 303, message = HttpStatuses.SEE_OTHER),
+        @ApiResponse(code = 200, message = HttpStatuses.OK, response = HabitAssignDto.class),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
         @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+        @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
     })
-    @PostMapping("/statistic/")
-    public ResponseEntity<Object> save(@Valid @RequestBody AddHabitStatisticDto addHabitStatisticDto) {
-        return ResponseEntity.status(HttpStatus.CREATED)
-            .body(habitStatisticService.save(addHabitStatisticDto));
-    }
-
-    /**
-     * Method for updating {@link HabitStatistic} by it's id.
-     *
-     * @param habitStatisticForUpdateDto - {@link UpdateHabitStatisticDto} with habit statistic id and
-     *                                   updated rate and amount of items.
-     * @return {@link UpdateHabitStatisticDto} instance.
-     */
-    @ApiOperation(value = "Update habit statistic.")
-    @ApiResponses(value = {
-        @ApiResponse(code = 201, message = HttpStatuses.CREATED),
-        @ApiResponse(code = 303, message = HttpStatuses.SEE_OTHER),
-        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
-    })
-    @PatchMapping("/statistic/{habitStatisticId}")
-    public ResponseEntity<UpdateHabitStatisticDto> updateStatistic(
-        @PathVariable Long habitStatisticId, @Valid @RequestBody UpdateHabitStatisticDto habitStatisticForUpdateDto) {
-        return ResponseEntity.status(HttpStatus.OK).body(habitStatisticService
-            .update(habitStatisticId, habitStatisticForUpdateDto));
+    @PatchMapping("/{id}/assign")
+    public ResponseEntity<HabitAssignDto> updateAssignByHabitId(
+        @ApiIgnore @CurrentUser UserVO userVO,
+        @PathVariable Long id, @Valid @RequestBody HabitAssignStatDto habitAssignStatDto) {
+        return ResponseEntity.status(HttpStatus.OK).body(habitAssignService
+            .updateStatusByHabitIdAndUserId(id, userVO.getId(), habitAssignStatDto));
     }
 
     /**
      * Method for finding all {@link HabitStatisticDto} by {@link Habit id}.
      *
-     * @param habitId {@link Habit} id.
+     * @param id {@link Habit} id.
```

Comment:
Looks like it makes sense to move all methods down to HabitStatisticsController

Reply:
Changes done.